rotating_proxies/middlewares.py

```diff
@@ -104,6 +107,8 @@ def engine_started(self):
         self.log_task.start(self.logstats_interval, now=True)
         self.reanimate_task = task.LoopingCall(self.reanimate_proxies)
         self.reanimate_task.start(self.reanimate_interval, now=False)
+        self.stats_task = task.LoopingCall(self.update_scrapystats)
+        self.stats_task.start(self.scrapystats_interval, now=True)
```

Comment:
I think it is better to stop this task in engine_stopped method.

Also, what do you think about updating stats when they changed, in _handle_result, instead of using a periodic task?

Reply:
I'll look into real time stats update; something was bothering me last time but now I don't see why `_handle_results` shouldn't be enough this time around.